modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/ViewRecycleBinSectionDisplayContext.java

```diff
@@ -73,9 +81,105 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems() {
 
 	@Override
 	protected String getCMSSectionFilterString() {
-		return "cmsKind eq 'object' and (cmsSection eq 'contents' or " +
-			"cmsSection eq 'files') and status eq " +
-				WorkflowConstants.STATUS_IN_TRASH;
+		String filter =
+			"cmsKind eq 'object' and (cmsSection eq 'contents' or cmsSection " +
+				"eq 'files') and status eq " +
+					WorkflowConstants.STATUS_IN_TRASH;
+
+		List<Long> eligible = new ArrayList<>();
+
+		try {
+			eligible = _getEligibleGroupIds(httpServletRequest);
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					"Unable to resolve eligible group ids", portalException);
+			}
+
+			return filter;
+		}
+
+		if (eligible.isEmpty()) {
+			return "cmsKind eq 'object' and (cmsSection eq 'contents' or " +
+				"cmsSection eq 'files') and status eq -1";
+		}
+
+		return filter + _buildGroupIdsAnyClause(eligible);
+	}
+
+	private String _buildGroupIdsAnyClause(List<Long> groupIds) {
```

Comment:
Can you simplify this logic for something like this?

```
	private String _buildGroupIdsAnyClause(List<Long> groupIds) {
		if (groupIds.isEmpty()) {
			return "";
		}

		StringBundler sb = new StringBundler();

		sb.append(" and ");

		if (groupIds.size() > 1) {
			sb.append("(");
		}

		boolean first = true;
		for (Long groupId : groupIds) {
			if (!first) {
				sb.append(" or ");
			}

			sb.append("groupIds/any(g:g eq ");
			sb.append(groupId);
			sb.append(")");

			first = false;
		}

		if (groupIds.size() > 1) {
			sb.append(")");
		}

		return sb.toString();
	}
```

Reply:
Never mind, I just realized that we can avoid many filters with OR by just using the IN clause, for example:

`roupIds/any(g:g in (37681,37682))`